<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>752</width>
    <height>538</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QString常用功能</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <widget class="QFrame" name="frame">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>271</width>
        <height>111</height>
       </rect>
      </property>
      <property name="title">
       <string>字符串截取</string>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>7</x>
         <y>39</y>
         <width>261</width>
         <height>71</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>front &amp; back</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="pushButton_3">
          <property name="text">
           <string>left &amp; right</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>first &amp; Last</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_4">
          <property name="text">
           <string>section *</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>140</y>
        <width>271</width>
        <height>121</height>
       </rect>
      </property>
      <property name="title">
       <string>存储相关</string>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <property name="geometry">
        <rect>
         <x>7</x>
         <y>39</y>
         <width>261</width>
         <height>71</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButton_25">
          <property name="text">
           <string>isNull &amp; isEmpty</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="pushButton_26">
          <property name="text">
           <string>resize</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushButton_27">
          <property name="text">
           <string>size &amp; length</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_28">
          <property name="text">
           <string>fill</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>270</y>
        <width>271</width>
        <height>121</height>
       </rect>
      </property>
      <property name="title">
       <string>搜索和判断</string>
      </property>
      <widget class="QWidget" name="layoutWidget_2">
       <property name="geometry">
        <rect>
         <x>7</x>
         <y>39</y>
         <width>261</width>
         <height>71</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButton_29">
          <property name="text">
           <string>indexOf &amp; lastIndexOf</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="pushButton_30">
          <property name="text">
           <string>contains</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushButton_31">
          <property name="text">
           <string>endsWith &amp; startsWith</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_32">
          <property name="text">
           <string>count</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_7">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>390</y>
        <width>271</width>
        <height>121</height>
       </rect>
      </property>
      <property name="title">
       <string>转换或修改</string>
      </property>
      <widget class="QWidget" name="layoutWidget_6">
       <property name="geometry">
        <rect>
         <x>7</x>
         <y>39</y>
         <width>261</width>
         <height>71</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_9">
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButton_45">
          <property name="text">
           <string>trimmed &amp; simplified</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="pushButton_46">
          <property name="text">
           <string>insert</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushButton_47">
          <property name="text">
           <string>remove</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_48">
          <property name="text">
           <string>replace</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_3">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QGroupBox" name="groupBox_8">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>331</width>
        <height>101</height>
       </rect>
      </property>
      <property name="title">
       <string>输入</string>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>54</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>str1</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>54</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>str2</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>30</y>
         <width>271</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox_2">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>70</y>
         <width>81</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>70</y>
         <width>51</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>参数</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox">
       <property name="geometry">
        <rect>
         <x>280</x>
         <y>70</y>
         <width>42</width>
         <height>24</height>
        </rect>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_9">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>140</y>
        <width>331</width>
        <height>351</height>
       </rect>
      </property>
      <property name="title">
       <string>结果显示</string>
      </property>
      <widget class="QPushButton" name="pushButton_5">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>40</y>
         <width>331</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>清空文本框</string>
       </property>
      </widget>
      <widget class="QPlainTextEdit" name="plainTextEdit">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>90</y>
         <width>331</width>
         <height>281</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
